### PR TITLE
Add Git hooks

### PR DESCRIPTION
**Pull request by huynhtrankhanh**

### CODE REVIEW

File: git-hooks/pre-commit [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
+python3 lean2md.py lean _posts
```
Comment (arthurpaulino) [inline git-hooks/pre-commit:2]: This is nice. We also need `git add _posts` right after that so the changes will be committed. Does it work?

Comment (arthurpaulino) [reply git-hooks/pre-commit:2]: Oh I just tested. It works!
```suggestion
python3 lean2md.py lean _posts
git add _posts
```

File: README.md [hunk 2]
```diff
@@ -13,3 +13,11 @@ You can place Lean code in the `lean` directory and run `$ python lean2md.py lea
 
 This repository is also a Lean project that has `mathlib` as dependency. You
 can get cached `olean` files by doing `leanproject get-mathlib-cache`.
+
+# Configure Git hooks
+This repository contains a precommit hook to automatically convert Lean source files to Markdown files on commit. If you want to use this hook, run
+```
+git config core.hooksPath ./git-hooks
```
Comment (arthurpaulino) [inline README.md]:
```suggestion
$ git config core.hooksPath ./git-hooks
$ chmod +x ./git-hooks/pre-commit
```

File: README.md [hunk 3]
```diff
@@ -13,3 +13,11 @@ You can place Lean code in the `lean` directory and run `$ python lean2md.py lea
 
 This repository is also a Lean project that has `mathlib` as dependency. You
 can get cached `olean` files by doing `leanproject get-mathlib-cache`.
+
+# Configure Git hooks
+This repository contains a precommit hook to automatically convert Lean source files to Markdown files on commit. If you want to use this hook, run
```
Comment (arthurpaulino) [inline README.md]:
```suggestion
This repository contains a pre-commit hook to automatically convert Lean source files to Markdown files on commit. If you want to use this hook, run
```

File: git-hooks/pre-commit (hunk 1 above)
Comment (arthurpaulino) [inline git-hooks/pre-commit:2]: I just tested it and it works!!
```suggestion
python3 lean2md.py lean _posts
git add _posts
```

**Review by arthurpaulino — CHANGES_REQUESTED** [review]

**Comment by Julian** [comment]
I was expecting it'd be pretty trivial to do this with GH Actions (and I believe it would be), so I'm not sure it's worth doing this "longer term" given that GH Actions can just regenerate stuff for us on push, but if you folks want it feel free to merge whenever satisfied.

**Comment by arthurpaulino** [comment]
@Julian do you mean making something (some GitHub bot?) commit the MD files on top of every push?

**Comment by Julian** [comment]
Yeah. Said bot could push to the same branch so it'd show up on the review

**Comment by arthurpaulino** [comment]
Oh, I've never configured anything like that. If you know how to do it then I think it's definitely an even cleaner solution!

**Comment by arthurpaulino** [comment]
The trade-off is: we'll need to pull everytime the bot commits if we want to change something. Do you think it's an issue? Sounds a bit annoying to me

**Comment by Julian** [comment]
Merge as is and we can always change I think

**Comment by arthurpaulino** [comment]
I probably misunderstood Julian's idea, but the way that [Eric explained](https://leanprover.zulipchat.com/#narrow/stream/113489-new-members/topic/Lean.20for.20the.20Inept.20Mathematician/near/267518975) looks good.

We wouldn't need to store lean and markdown files of the same post.

**Pull request merged**